src/Decomposition.C

```diff
@@ -248,6 +294,33 @@ int OctDecomposition::flush(std::vector<Particle> &particles, const SendParticle
   return flush_count;
 }
 
+void OctDecomposition::countAssignments(const std::vector<GenericSplitter>& states, const std::vector<Particle>& particles, Reader* reader, const CkCallback& cb) {
+  std::vector<int> counts (states.size()/2, 0);
+
+  // Search for the first particle whose key is greater or equal to the input key,
+  // in the range [start, finish). This should also work for OCT as the particle
+  // keys are SFC keys.
+  int start = 0;
+  int finish = particles.size();
+  Key from, to;
+  std::function<bool(const Particle&, Key)> compGE = [] (const Particle& a, Key b) {return a.key >= b;};
+  if (particles.size() > 0) {
+    for (int i = 0; i < counts.size(); i++) {
+      from = states[2*i].split_key;
```

Comment:
Change this split_key nonsense to start_key and end_key and then states and keys will be the same size.